### PR TITLE
docs(storybook): add a11y addon

### PR DESCRIPTION
**Pull request by marielaures**
**Summary**

Add a new Storybook addon to monitor our components for being accessible.
See #221.

**Result**

There is a new "Accessibility" tab displayed on the right sidebar that shows both the a11y "violations" and the "passes". 

image

### CODE REVIEW

**Comment by Haroenv** [comment]
@marielaures did you know that you can just leave the merge conflicts in a yarn.lock and run `yarn`? it will fix the conflicts automatically

**Comment by vvo** [comment]
> @marielaures did you know that you can just leave the merge conflicts in a yarn.lock and run yarn? it will fix the conflicts automatically

Can you explain how you would fix the current conflict?

I don't understand right now why there are conflicts o_O I tried to fix them too by doing a git checkout from master and then installing the module, but it does not change anything, weiiird.

**Comment by mthuret** [comment]
@vvo I'm wondering if it's not due to renovate PR that are merged without waiting for the robot to merge previous one before. 

If that's a case, let's unprotect master and have automerge. WDYF @samouss?

**Comment by vvo** [comment]
> @vvo I'm wondering if it's not due to renovate PR that are merged without waiting for the robot to merge previous one before.

Not sure to understand what you mean

**Comment by Haroenv** [comment]
I just saw in the GitHub interface here that there's a conflict in that file. 

```sh
$ git co master
$ git pull
$ git co this-branch
$ git rebase master
bla bra there's a conflict
$ yarn
```

If I remember it correctly from last time I had this issue, that's what I meant

File: package.json [hunk 1]
```diff
@@ -143,6 +143,7 @@
   },
   "dependencies": {
     "algolia-aerial": "^1.1.4",
-    "algolia-frontend-components": "^0.0.34"
+    "algolia-frontend-components": "^0.0.34",
+    "storybook-addon-a11y": "^3.1.9"
```
Comment (samouss) [inline package.json]: Should probably be a `devDependencies`

**Comment by samouss** [comment]
Since we always call something like:

```js
const stories = storiesOf('Name of story', module);

stories
  .addDecorator(XX)
  .addDecorator(YY);
```

Why don't we wrap the call to storiesOf? It will simplify the process of adding more decorator in the future. Not a strong opinion on it, neither if it's a good idea.

File: package.json (hunk 1 above)
Comment (vvo) [reply package.json]: done

**Comment by vvo** [comment]
Branch should be fixed, here's what went wrong (only for full transparency and understanding, not finger pointing anyone):
- @marielaures started a branch, worked on it
- other commits were merged into master, some of them changing package.json
- @marielaures pushed to GH
- GH showed issues with yarn.lock file (conflicts)
- We tried to fix it, the wrong way (checkout yarn.lock file from master, relaunch yarn in repo), my bad here!
- @marielaures tried to fix it by deleting yarn.lock file, triggered more issues because then dependencies were updated to different versions
- To fix it well, next time, do what @Haroenv suggested:

```
git co master
git pull
git co this-branch
git rebase master
# conflict
yarn
git add yarn.lock
git rebase --continue
```

**Comment by vvo** [comment]
Any idea what:

image

is? @samouss/

**Comment by Haroenv** [comment]
sw_vers seems to be some macOS-only way of getting the version of the OS

**Comment by samouss** [comment]
@vvo Nope no ideas! But as @Haroenv said, seems more related to the OS than our build. I try to look at the history of the builds but I didn't find real clues.

**Comment by marielaures** [comment]
Will wrap the call to 'storiesOf' in a separate PR @samouss!

File: stories/Highlight.stories.js [hunk 2]
```diff
@@ -48,6 +49,7 @@ StrongHits.propTypes = {
 };
 
 stories
+  .addDecorator(checkA11y)
```
Comment (samouss) [inline stories/Highlight.stories.js]: It have been added twice?

**Comment by Haroenv** [comment]
Congrats @marielaures

**Comment by vvo** [comment]
GG @marielaures

**Pull request merged**